src/ckeditor.tsx

```diff
@@ -180,9 +183,66 @@ export default class CKEditor<TEditor extends Editor> extends React.Component<Pr
 				const viewDocument = editor.editing.view.document;
 
 				modelDocument.on<DocumentChangeEvent>( 'change:data', event => {
+					const changedRoots: ChangedRoots = {};
+
+					modelDocument.differ.getChanges()
+						.forEach( change => {
+							if ( 'position' in change && change.position.root.isAttached() ) {
```

Comment:
As far as I remember attribute change diff items contain only information about changing the state of the root: Attached/Detached. It was useless for me in that place, since I just wanted to get the information on whether the data changed. There is another callback that handles detached roots.

Reply:
I meant the element and text node attributes, like `bold`, `alignment`, `headingRows` etc.